### PR TITLE
Implement save form feature

### PR DESCRIPTION
**Pull request by kumuditha-udayanga**
### Purpose
- Implement form saving feature to esm-formbuilder
- Update User Interface
- This enables users to 
  - Create, update or save a form as a new version.
  - Create, edit schemas and upload to a form.
  - Publish/ Unpublish forms.
  
 ### Screenshots
Screenshot 2022-08-25 at 11 12 52 PM
Screenshot 2022-08-25 at 1 18 21 PM
Screenshot 2022-08-25 at 1 18 27 PM

#### Demo on creating new forms
https://vimeo.com/738267453

### CODE REVIEW

**Comment by jonathandick** [comment]
@kumuditha-udayanga could you possibly update CI to deploy this to dev3?

**Comment by kumuditha-udayanga** [comment]
@jonathandick, I updated the deployment process

**Comment by kumuditha-udayanga** [comment]
Hi @samuelmale thank you for reviewing, I updated the PR, attached few screenshots of the changes and a screen record link.

**Comment by samuelmale** [comment]
@kumuditha-udayanga don't you think the form name can be inferred from the form json?


Screenshot 2022-08-26 at 09 07 15

File: src/components/form-editor/modals/save-form.tsx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+      setOpenSaveFormModal(true);
+    } else if (option === "new") {
+      setSaveState("newVersion");
+      setOpenSaveFormModal(true);
+    } else if (option === "update") {
+      setSaveState("update");
+      setOpenConfirmSaveModal(false);
+      setOpenSaveFormModal(true);
+    }
+  };
```
Comment (samuelmale) [inline src/components/form-editor/modals/save-form.tsx:67]: I would wrap this in a `useCallback`

File: src/context/context.tsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { createContext } from "react";
+
+export const SchemaContext = createContext(null);
```
Comment (samuelmale) [inline src/context/context.tsx:3]: Can you define typings for the `SchemaContext`?

**Review by samuelmale — APPROVED** [review]
Nice work, just a few minor comments

**Pull request merged**